terraform/context.go

```diff
@@ -253,7 +261,7 @@ func (c *Context) Apply() (*State, error) {
 //
 // Plan also updates the diff of this context to be the diff generated
 // by the plan, so Apply can be called after.
-func (c *Context) Plan(opts *PlanOpts) (*Plan, error) {
+func (c *Context) Plan() (*Plan, error) {
```

Comment:
Hm, so the reason we had `*PlanOpts` initially was that I expected targeted applies/partial to go in here and to be part of the plan itself. I'm not deeper into the review yet, but I think that targeted ops should be a part of the plan, so that `Apply` doesn't actually know about it (the plan is just saying that we're only doing operations on certain things). 

I think the UX should be if you do a `terraform plan -target=foo -out=file.tflan`, then you can do a `terraform apply file.tfplan` and you only affect `foo`. Is this the case currently?